<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>750</width>
    <height>506</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>上海交通大学&quot;交龙&quot;智能轮椅导航平台</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <property name="layoutDirection">
    <enum>Qt::LeftToRight</enum>
   </property>
   <layout class="QGridLayout" name="gridLayout_2">
    <item row="0" column="0" rowspan="3">
     <widget class="GraphWidget" name="graphicsView">
      <property name="minimumSize">
       <size>
        <width>400</width>
        <height>450</height>
       </size>
      </property>
     </widget>
    </item>
    <item row="0" column="1">
     <widget class="laserWidget" name="laserDisplay" native="true">
      <property name="minimumSize">
       <size>
        <width>187</width>
        <height>151</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>187</width>
        <height>20000</height>
       </size>
      </property>
     </widget>
    </item>
    <item row="1" column="1">
     <widget class="QGroupBox" name="status">
      <property name="minimumSize">
       <size>
        <width>187</width>
        <height>40</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>187</width>
        <height>160</height>
       </size>
      </property>
      <property name="title">
       <string/>
      </property>
      <layout class="QGridLayout" name="gridLayout">
       <item row="1" column="0">
        <widget class="QCheckBox" name="setFocusOnRobot">
         <property name="font">
          <font>
           <pointsize>14</pointsize>
           <weight>50</weight>
           <italic>false</italic>
           <bold>false</bold>
          </font>
         </property>
         <property name="text">
          <string>视角跟随机器人</string>
         </property>
        </widget>
       </item>
       <item row="0" column="0">
        <widget class="QCheckBox" name="setHistoryRobot">
         <property name="font">
          <font>
           <pointsize>14</pointsize>
          </font>
         </property>
         <property name="acceptDrops">
          <bool>false</bool>
         </property>
         <property name="text">
          <string>显示机器人轨迹</string>
         </property>
         <property name="tristate">
          <bool>false</bool>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item row="2" column="1">
     <layout class="QVBoxLayout" name="verticalLayout_2">
      <item>
       <widget class="QComboBox" name="comboBox">
        <property name="minimumSize">
         <size>
          <width>185</width>
          <height>35</height>
         </size>
        </property>
        <property name="font">
         <font>
          <pointsize>14</pointsize>
         </font>
        </property>
        <item>
         <property name="text">
          <string>自动模式</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>半自动模式</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>手动模式</string>
         </property>
        </item>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout">
        <item>
         <widget class="QPushButton" name="setGoal">
          <property name="minimumSize">
           <size>
            <width>0</width>
            <height>32</height>
           </size>
          </property>
          <property name="font">
           <font>
            <pointsize>12</pointsize>
           </font>
          </property>
          <property name="text">
           <string>设定目标</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="setPos">
          <property name="minimumSize">
           <size>
            <width>0</width>
            <height>32</height>
           </size>
          </property>
          <property name="font">
           <font>
            <pointsize>12</pointsize>
           </font>
          </property>
          <property name="acceptDrops">
           <bool>true</bool>
          </property>
          <property name="autoFillBackground">
           <bool>true</bool>
          </property>
          <property name="text">
           <string>设定初始点</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QPushButton" name="setTopo">
        <property name="minimumSize">
         <size>
          <width>185</width>
          <height>32</height>
         </size>
        </property>
        <property name="font">
         <font>
          <pointsize>12</pointsize>
         </font>
        </property>
        <property name="text">
         <string>执行选定任务</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="startRobot">
        <property name="minimumSize">
         <size>
          <width>185</width>
          <height>32</height>
         </size>
        </property>
        <property name="font">
         <font>
          <pointsize>12</pointsize>
         </font>
        </property>
        <property name="text">
         <string>启动机器人</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="stopRobot">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="minimumSize">
         <size>
          <width>185</width>
          <height>32</height>
         </size>
        </property>
        <property name="font">
         <font>
          <pointsize>12</pointsize>
         </font>
        </property>
        <property name="text">
         <string>停止机器人</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
   <widget class="QDial" name="dial">
    <property name="geometry">
     <rect>
      <x>410</x>
      <y>340</y>
      <width>100</width>
      <height>100</height>
     </rect>
    </property>
    <property name="minimumSize">
     <size>
      <width>100</width>
      <height>100</height>
     </size>
    </property>
    <property name="maximumSize">
     <size>
      <width>100</width>
      <height>100</height>
     </size>
    </property>
    <property name="minimum">
     <number>-100</number>
    </property>
    <property name="maximum">
     <number>100</number>
    </property>
    <property name="singleStep">
     <number>1</number>
    </property>
    <property name="pageStep">
     <number>10</number>
    </property>
    <property name="value">
     <number>0</number>
    </property>
    <property name="invertedAppearance">
     <bool>false</bool>
    </property>
    <property name="wrapping">
     <bool>false</bool>
    </property>
    <property name="notchesVisible">
     <bool>true</bool>
    </property>
   </widget>
   <zorder>graphicsView</zorder>
   <zorder>laserDisplay</zorder>
   <zorder>status</zorder>
   <zorder>dial</zorder>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>750</width>
     <height>28</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuHost">
    <property name="title">
     <string>设置</string>
    </property>
    <addaction name="action_2"/>
   </widget>
   <addaction name="menuHost"/>
  </widget>
  <action name="action">
   <property name="text">
    <string>文件</string>
   </property>
  </action>
  <action name="action_2">
   <property name="text">
    <string>地址</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>GraphWidget</class>
   <extends>QGraphicsView</extends>
   <header location="global">component</header>
   <slots>
    <slot>zoom(int)</slot>
    <slot>enterSetPose()</slot>
    <slot>enterSetGoal()</slot>
    <slot>setFocusOnRobot(bool)</slot>
    <slot>setShowHistoryRobot(bool)</slot>
   </slots>
  </customwidget>
  <customwidget>
   <class>laserWidget</class>
   <extends>QWidget</extends>
   <header>component</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>dial</sender>
   <signal>valueChanged(int)</signal>
   <receiver>graphicsView</receiver>
   <slot>zoom(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>499</x>
     <y>484</y>
    </hint>
    <hint type="destinationlabel">
     <x>303</x>
     <y>301</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>startRobot</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>startGoal()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>697</x>
     <y>521</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>stopRobot</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>stopGoal()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>697</x>
     <y>554</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>setGoal</sender>
   <signal>clicked()</signal>
   <receiver>graphicsView</receiver>
   <slot>enterSetGoal()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>648</x>
     <y>448</y>
    </hint>
    <hint type="destinationlabel">
     <x>303</x>
     <y>301</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>setPos</sender>
   <signal>clicked()</signal>
   <receiver>graphicsView</receiver>
   <slot>enterSetPose()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>742</x>
     <y>448</y>
    </hint>
    <hint type="destinationlabel">
     <x>303</x>
     <y>301</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>setTopo</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>startTopoTask()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>697</x>
     <y>510</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>setFocusOnRobot</sender>
   <signal>toggled(bool)</signal>
   <receiver>graphicsView</receiver>
   <slot>setFocusOnRobot(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>703</x>
     <y>396</y>
    </hint>
    <hint type="destinationlabel">
     <x>304</x>
     <y>320</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>setHistoryRobot</sender>
   <signal>toggled(bool)</signal>
   <receiver>graphicsView</receiver>
   <slot>setShowHistoryRobot(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>703</x>
     <y>368</y>
    </hint>
    <hint type="destinationlabel">
     <x>304</x>
     <y>325</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_2</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>setHost()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>374</x>
     <y>252</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>startGoal()</slot>
  <slot>stopGoal()</slot>
  <slot>startTopoTask()</slot>
  <slot>setHost()</slot>
 </slots>
</ui>
